ci: dask nightlies failing due to ordering assumptions in tests

we're getting these failures with dask nightlies:
```python
FAILED tests/expr_and_series/over_test.py::test_over_single[dask] - AssertionError: Mismatch at index 0: b != a
Expected: {'a': ['a', 'a', 'b', 'b', 'b'], 'b': [1, 2, 3, 5, 3], 'c': [5, 4, 3, 2, 1], 'c_max': [5, 5, 3, 3, 3]}
Got: {'a': ['b', 'b', 'a', 'a', 'b'], 'b': [3, 5, 1, 2, 3], 'c': [3, 2, 5, 4, 1], 'c_max': [3, 3, 5, 5, 3]}
FAILED tests/expr_and_series/over_test.py::test_over_multiple[dask] - AssertionError: Mismatch at index 0: b != a
Expected: {'a': ['a', 'a', 'b', 'b', 'b'], 'b': [1, 2, 3, 5, 3], 'c': [5, 4, 3, 2, 1]}
Got: {'a': ['b', 'b', 'a', 'a', 'b'], 'b': [3, 5, 1, 2, 3], 'c': [3, 2, 5, 4, 1]}
FAILED tests/expr_and_series/over_test.py::test_over_std_var[dask] - AssertionError: Mismatch at index 0: b != a
Expected: {'a': ['a', 'a', 'b', 'b', 'b'], 'b': [1, 2, 3, 5, 3], 'c': [5, 4, 3, 2, 1], 'c_std0': [0.5, 0.5, 0.816496580927726, 0.816496580927726, 0.816496580927726], 'c_std1': [0.7071067811865476, 0.7071067811865476, 1.0, 1.0, 1.0], 'c_var0': [0.25, 0.25, 0.6666666666666666, 0.6666666666666666, 0.6666666666666666], 'c_var1': [0.5, 0.5, 1.0, 1.0, 1.0]}
Got: {'a': ['b', 'b', 'a', 'a', 'b'], 'b': [3, 5, 1, 2, 3], 'c': [3, 2, 5, 4, 1], 'c_std0': [0.816496580927726, 0.816496580927726, 0.5, 0.5, 0.816496580927726], 'c_std1': [1.0, 1.0, 0.7071067811865476, 0.7071067811865476, 1.0], 'c_var0': [0.6666666666666666, 0.6666666666666666, 0.25, 0.25, 0.6666666666666666], 'c_var1': [1.0, 1.0, 0.5, 0.5, 1.0]}
```

I think we just need to add a `sort` in those tests to make the test not order-dependent

--- a/narwhals/dataframe.py
+++ b/narwhals/dataframe.py
@@ -2176,7 +2176,8 @@ class LazyFrame(BaseFrame[FrameT]):
                     "Hints:\n"
                     "- Instead of `lf.select(nw.col('a').sort())`, use `lf.select('a').sort()`.\n"
                     "- Instead of `lf.select(nw.col('a').cum_sum())`, use\n"
-                    "  `lf.select(nw.col('a').cum_sum().over(order_by='date'))`."
+                    "  `lf.select(nw.col('a').cum_sum().over(order_by='date'))`.\n\n"
+                    "See https://narwhals-dev.github.io/narwhals/basics/order_dependence/."
                 )
                 raise OrderDependentExprError(msg)
             if arg._metadata.kind.is_filtration():

--- a/narwhals/dataframe.py
+++ b/narwhals/dataframe.py
@@ -2174,11 +2174,9 @@ class LazyFrame(BaseFrame[FrameT]):
                 msg = (
                     "Order-dependent expressions are not supported for use in LazyFrame.\n\n"
                     "Hints:\n"
-                    "- Instead of `lf.select(nw.col('a').sort())`, use `lf.select('a').sort()\n"
-                    "- Instead of `lf.select(nw.col('a').head())`, use `lf.select('a').head()\n"
-                    "- `Expr.cum_sum`, and other such expressions, are not currently supported.\n"
-                    "  In a future version of Narwhals, a `order_by` argument will be added to\n"
-                    "  `over` and they will be supported."
+                    "- Instead of `lf.select(nw.col('a').sort())`, use `lf.select('a').sort()`.\n"
+                    "- Instead of `lf.select(nw.col('a').cum_sum())`, use\n"
+                    "  `lf.select(nw.col('a').cum_sum().over(order_by='date'))`."
                 )
                 raise OrderDependentExprError(msg)
             if arg._metadata.kind.is_filtration():

--- a/tests/expr_and_series/arithmetic_test.py
+++ b/tests/expr_and_series/arithmetic_test.py
@@ -9,6 +9,7 @@ from hypothesis import given
 
 import narwhals.stable.v1 as nw
 from tests.utils import DASK_VERSION
+from tests.utils import DUCKDB_VERSION
 from tests.utils import PANDAS_VERSION
 from tests.utils import Constructor
 from tests.utils import ConstructorEager
@@ -341,7 +342,9 @@ def test_arithmetic_series_left_literal(
 
 
 def test_std_broadcating(constructor: Constructor) -> None:
-    # `std(ddof=2)` fails for duckdb here
+    if "duckdb" in str(constructor) and DUCKDB_VERSION < (1, 3):
+        # `std(ddof=2)` fails for duckdb here
+        pytest.skip()
     df = nw.from_native(constructor({"a": [1, 2, 3]}))
     result = df.with_columns(b=nw.col("a").std()).sort("a")
     expected = {"a": [1, 2, 3], "b": [1.0, 1.0, 1.0]}

--- a/tests/expr_and_series/over_test.py
+++ b/tests/expr_and_series/over_test.py
@@ -40,12 +40,13 @@ def test_over_single(constructor: Constructor) -> None:
         "a": ["a", "a", "b", "b", "b"],
         "b": [1, 2, 3, 5, 3],
         "c": [5, 4, 3, 2, 1],
+        "i": list(range(5)),
         "c_max": [5, 5, 3, 3, 3],
     }
 
-    result = df.with_columns(c_max=nw.col("c").max().over("a")).sort("i").drop("i")
+    result = df.with_columns(c_max=nw.col("c").max().over("a")).sort("i")
     assert_equal_data(result, expected)
-    result = df.with_columns(c_max=nw.col("c").max().over(["a"])).sort("i").drop("i")
+    result = df.with_columns(c_max=nw.col("c").max().over(["a"])).sort("i")
     assert_equal_data(result, expected)
 
 
@@ -61,6 +62,7 @@ def test_over_std_var(request: pytest.FixtureRequest, constructor: Constructor) 
         "a": ["a", "a", "b", "b", "b"],
         "b": [1, 2, 3, 5, 3],
         "c": [5, 4, 3, 2, 1],
+        "i": list(range(5)),
         "c_std0": [0.5, 0.5, 0.816496580927726, 0.816496580927726, 0.816496580927726],
         "c_std1": [0.7071067811865476, 0.7071067811865476, 1.0, 1.0, 1.0],
         "c_var0": [
@@ -73,16 +75,12 @@ def test_over_std_var(request: pytest.FixtureRequest, constructor: Constructor) 
         "c_var1": [0.5, 0.5, 1.0, 1.0, 1.0],
     }
 
-    result = (
-        df.with_columns(
-            c_std0=nw.col("c").std(ddof=0).over("a"),
-            c_std1=nw.col("c").std(ddof=1).over("a"),
-            c_var0=nw.col("c").var(ddof=0).over("a"),
-            c_var1=nw.col("c").var(ddof=1).over("a"),
-        )
-        .sort("i")
-        .drop("i")
-    )
+    result = df.with_columns(
+        c_std0=nw.col("c").std(ddof=0).over("a"),
+        c_std1=nw.col("c").std(ddof=1).over("a"),
+        c_var0=nw.col("c").var(ddof=0).over("a"),
+        c_var1=nw.col("c").var(ddof=1).over("a"),
+    ).sort("i")
     assert_equal_data(result, expected)
 
 
@@ -94,6 +92,7 @@ def test_over_multiple(constructor: Constructor) -> None:
         "a": ["a", "a", "b", "b", "b"],
         "b": [1, 2, 3, 3, 5],
         "c": [5, 4, 3, 1, 2],
+        "i": list(range(5)),
         "c_min": [5, 4, 1, 1, 2],
     }
     expected = {
@@ -102,7 +101,7 @@ def test_over_multiple(constructor: Constructor) -> None:
         "c": [5, 4, 3, 2, 1],
     }
 
-    result = df.with_columns(c_min=nw.col("c").min().over("a", "b")).sort("i").drop("i")
+    result = df.with_columns(c_min=nw.col("c").min().over("a", "b")).sort("i")
     assert_equal_data(result, expected)
 
 

--- a/tests/expr_and_series/over_test.py
+++ b/tests/expr_and_series/over_test.py
@@ -53,6 +53,8 @@ def test_over_std_var(request: pytest.FixtureRequest, constructor: Constructor) 
     if "cudf" in str(constructor):
         # https://github.com/rapidsai/cudf/issues/18159
         request.applymarker(pytest.mark.xfail)
+    if "duckdb" in str(constructor) and DUCKDB_VERSION < (1, 3):
+        pytest.skip()
 
     df = nw.from_native(constructor(data))
     expected = {

--- a/tests/frame/add_test.py
+++ b/tests/frame/add_test.py
@@ -1,11 +1,16 @@
 from __future__ import annotations
 
+import pytest
+
 import narwhals.stable.v1 as nw
+from tests.utils import DUCKDB_VERSION
 from tests.utils import Constructor
 from tests.utils import assert_equal_data
 
 
 def test_add(constructor: Constructor) -> None:
+    if "duckdb" in str(constructor) and DUCKDB_VERSION < (1, 3):
+        pytest.skip()
     data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8.0, 9.0]}
     df = nw.from_native(constructor(data))
     result = df.with_columns(

